homeassistant/components/rituals_perfume_genie/switch.py

```diff
@@ -0,0 +1,95 @@
+"""Support for Rituals Perfume Genie switches."""
+from datetime import timedelta
+
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN
+
+SCAN_INTERVAL = timedelta(seconds=30)
+
+ON_STATE = "1"
+AVAILABLE_STATE = 1
+
+MANUFACTURER = "Rituals Cosmetics"
+MODEL = "Diffuser"
+ICON = "mdi:fan"
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the diffuser switch."""
+    session = async_get_clientsession(hass)
+    account = hass.data[DOMAIN][config_entry.entry_id]
+    diffusers = await account.get_devices(session)
```

Comment:
It would be a bit cleaner if every call didn't have to re-pass the session object

Reply:
This has been fixed in https://github.com/home-assistant/core/pull/46218/commits/b61d74e9bd7e7abd60dabcc9983090769f2ac4ec